dotCMS/src/main/java/com/dotcms/saml/DotIdentityProviderConfigurationImpl.java

```diff
@@ -0,0 +1,128 @@
+package com.dotcms.saml;
+
+import com.dotcms.saml.service.external.IdentityProviderConfiguration;
+import com.dotcms.saml.service.external.SamlName;
+import com.dotcms.security.apps.AppSecrets;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotcms.security.apps.Secret;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import io.vavr.control.Try;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Optional;
+
+/**
+ * Default implementation to retrieve the configuration from apps
+ * @author jsanca
+ */
+public class DotIdentityProviderConfigurationImpl implements IdentityProviderConfiguration {
+
+    private final AppsAPI appsAPI;
+    private final Host    host;
+
+    public DotIdentityProviderConfigurationImpl(final AppsAPI appsAPI, final Host host) {
+
+        this.appsAPI = appsAPI;
+        this.host    = host;
+
+    }
+
+    private final Optional<Secret> findSecret (final String key) {
+
+        final Optional<AppSecrets> appSecretOpt =
+                Try.of(()->this.appsAPI.getSecrets(DotSamlConfigurationServiceImpl.DOT_SAML_DEFAULT_PROPERTIES_CONTEXT_MAP_KEY,
+                        true, host, APILocator.systemUser())).getOrElseGet(e -> Optional.empty());
+
+        return appSecretOpt.isPresent() && appSecretOpt.get().getSecrets().containsKey(key)?
+                Optional.of(appSecretOpt.get().getSecrets().get(key)): Optional.empty();
+    }
+
+    @Override
+    public boolean isEnabled() {
+
+
+        final String enableKey           = SamlName.DOT_SAML_ENABLE.getPropertyName();
+        final Optional<Secret> secretOpt = this.findSecret(enableKey);
+        return secretOpt.isPresent()? secretOpt.get().getBoolean(): false;
```

Comment:
Codacy found an issue: [Ternary operators that can be simplified with || or &&](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)